python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -54,16 +56,23 @@ public void initialize(Context context) {
       }
       Set<String> interpolationIdentifiers = extractStringInterpolationIdentifiers(functionTree);
       for (TreeSymbol symbol : functionTree.localVariables()) {
-        List<Tree> usages = symbol.usages().stream()
-          .filter(tree -> !tree.parent().is(Kind.PARAMETER))
-          .collect(Collectors.toList());
-        if (interpolationIdentifiers.stream().noneMatch(id -> id.contains(symbol.name())) && !"_".equals(symbol.name()) && usages.size() == 1) {
-          symbol.usages().forEach(usage -> ctx.addIssue(usage, String.format(MESSAGE, symbol.name())));
+        if (interpolationIdentifiers.stream().noneMatch(id -> id.contains(symbol.name())) && !"_".equals(symbol.name()) && symbol.usages().size() == 1) {
+          symbol.usages().stream()
+            .filter(tree -> tree.parent() == null || !tree.parent().is(Kind.PARAMETER))
+            .filter(tree -> !isTupleDeclaration(tree))
+            .forEach(usage -> ctx.addIssue(usage, String.format(MESSAGE, symbol.name())));
         }
       }
     });
   }
 
+  private static boolean isTupleDeclaration(Tree tree) {
+    return tree.ancestors().stream()
+      .anyMatch(t -> t.is(Kind.TUPLE)
+        || (t.is(Kind.EXPRESSION_LIST) && ((PyExpressionListTree) t).expressions().size() > 1)
+        || t.is(Kind.FOR_STMT) && ((PyForStatementTree) t).expressions().size() > 1 && ((PyForStatementTree) t).expressions().contains(tree));
```

Comment:
Shouldn't we have tuples in all cases?

Reply:
Tuple would be an ancestor before `FOR_STMT` and `EXPRESSION_LIST`, so it will return `true` for the first condition. Adding a test for it